addons/web_editor/static/lib/odoo-editor/test/spec/list.test.js

```diff
@@ -3018,20 +3018,20 @@ describe('List', () => {
                             await testEditor(BasicEditor, {
                                 contentBefore: '<ol><li>abc</li><li>[]def</li></ol>',
                                 stepFunction: deleteBackward,
-                                contentAfter: '<ol><li>abc[]def</li></ol>',
+                                contentAfter: '<ol><li>abc</li></ol><p>[]def</p>',
```

Comment:
@Zinston This was weird to me as well at first, but in fact matches Google Docs.

Reply:
Wow this seems completely contrary to the approach above though... I guess the heuristic is not always to be trusted? Again a matter of choice but this to me is inconsistent with the other choices made.